djangocms_versioning/managers.py

```diff
@@ -1,14 +1,39 @@
+from django.db import models
+from django.db.models.sql.where import WhereNode
+
 from .constants import PUBLISHED
 
 
+class PublishedQuerySet(models.QuerySet):
+
+    def get(self, *args, **kwargs):
+        """
+        A get query should always be able to access entries which are not in
+        a published state.
+        """
+        where_children = self.query.where.children
+        all_except_published = [
```

Comment:
@jonathan-s What is the benefit of this approach over an annotated query?

Reply:
I would have thought that the query would need to be processed, returned and then looped through via Python before potentially being filtered further requiring another trip to the DB as any query on Page or Filer would hammer the .get method.

Reply:
From my understanding when you make an annotated query (ie `.annotate(...)`) you're adding extra data for the `SELECT` part of the query. Since the query is filtered in the `WHERE` part (ie `WHERE "djangocms_versioning_version"."state" = 'published'`) this is what we want to get rid of. Using annotate won't solve that. 

> I would have thought that the query would need to be processed, returned and then looped through via Python before potentially being filtered further requiring another trip to the DB as any query on Page or Filer would hammer the .get method.

Not sure what exactly you're after here. Do you think you could clarify if the above doesn't already answer your question.